python/dgl/heterograph.py

```diff
@@ -727,24 +730,35 @@ def ndata(self):
         >>> g = dgl.graph([(0, 1), (1, 2)], 'user', 'follows')
         >>> g.ndata['h'] = torch.zeros(3, 5)
 
+        To set features of all nodes in a heterogeneous graph
+        with multiple node types:
+
+        >>> g = dgl.graph([(0, 1), (1, 2), (1, 0)], 'user', 'like', 'movie')
+        >>> g.ndata['h'] = {'user': torch.zeros(2, 5),
+                            'movie': torch.zeros(3, 5)}
+
         See Also
         --------
         nodes
         """
-        ntid = self.get_ntype_id(None)
-        ntype = self.ntypes[0]
-        return HeteroNodeDataView(self, ntype, ntid, ALL)
+        if len(self.ntypes) == 1:
+            ntid = self.get_ntype_id(None)
+            ntype = self.ntypes[0]
+            return HeteroNodeDataView(self, ntype, ntid, ALL)
+        else:
+            ntids = [self.get_ntype_id(ntype) for ntype in self.ntypes]
+            ntypes = self.ntypes
+            return HeteroNodeDataView(self, ntypes, ntids, ALL)
```

Comment:
I feel the codes can be simplified. Here, you just do
```python
ntids = [self.get_ntype_id(ntype) for ntype in self.ntypes]
ntypes = self.ntypes
return HeteroNodeDataView(self, ntypes, ntids, ALL)
```
no matter what. And in `HeteroNodeDataView`, you retrieve the data using the `ntids` and feature name. Before you return the data, check if the dictionary contains only one item and return the value if so.

Reply:
> ```python
> HeteroNodeDataView
> ```

This maybe in another PR. A lots of place use HeteroNodeDataView.

Reply:
ok